data/scripts/set_wallpaper

```diff
@@ -227,17 +227,48 @@ fi
 # For simple WMs, use either feh or nitrogen
 # Implementation note: this uses spaces around list items to enforce matching whole words.
 # This also means that an empty variable won't cause false positives, since it expands to "  "
-SIMPLE_WMS=("bspwm" "dwm" "herbstluftwm" "i3" "i3-gnome" "i3-with-shmlog" "jwm" "jdwm" "LeftWM" "openbox" "qtile" "qtile-venv" "spectrwm" "xmonad")
-if [[ " ${SIMPLE_WMS[*]} " = *" $XDG_CURRENT_DESKTOP "* || " ${SIMPLE_WMS[*]} " = *" $XDG_SESSION_DESKTOP "* ||
-      " ${SIMPLE_WMS[*]} " = *" $DESKTOP_SESSION "* ]]; then
-	if command -v "feh" >/dev/null 2>&1; then
-	      # TODO: This should take the scaling parameter $4 into account and use other options than --bg-fill
-        feh --bg-fill "$WP" 2> /dev/null
+
+SIMPLE_WMS=(
+  "bspwm"
+  "dwm"
+  "herbstluftwm"
+  "i3"
+  "i3-gnome"
+  "i3-with-shmlog"
+  "jwm"
+  "jdwm"
+  "LeftWM"
+  "openbox"
+  "qtile"
+  "qtile-venv"
+  "spectrwm"
+  "xmonad"
+)
+
+# Get a unique list (usually one) of possible simple window managers
+# The syntax for these is <desktop>+<window_manager>.
+#
+# Simple window managers usually have no desktop environment
+# so the init script should set XDG_CURRENT_DESKTOP to "none+i3" or similar.
```

Comment:
Sorry, is this actually an established convention? I don't see the "none+" mentioned in https://wiki.archlinux.org/title/Xdg-utils#Environment_variables for instance, and I find it awkwardly redundant.

Reply:
My understanding is it's part of the display manager and specific to nixos. You're idea to try setting with feh or nitrogen would mean we don't have to know any of this so probably that is the route to take